src/NLog/Layouts/SimpleLayout.cs

```diff
@@ -306,13 +306,12 @@ public override void Precalculate(LogEventInfo logEvent)
                 {
                     // If raw value doesn't have the ability to mutate, then we can skip precalculate
                     var success = _rawValueRenderer.TryGetRawValue(logEvent, out var value);
-                    if (success && Convert.GetTypeCode(value) != TypeCode.Object || value.GetType().IsValueType())
+                    if (success && (Convert.GetTypeCode(value) != TypeCode.Object || value.GetType().IsValueType()))
```

Comment:
proposal, check on `null` explicitly:

```
  if (success && value != null && (Convert.GetTypeCode(value) != TypeCode.Object || value.GetType().IsValueType()))
```

Reply:
`Convert.GetTypeCode(null)` returns [TypeCode.Empty](https://docs.microsoft.com/en-us/dotnet/api/system.typecode), so extra check not needed.

Reply:
it's needed for the readability ;)

It's a bit hidden that there isn't a nullref exception with `value.GetType()` in this case. Also the overhead of the method call is a bit of waste

Reply:
added it :angel: